src/index.js

```diff
@@ -209,6 +216,10 @@ export class BrowserHistory extends History {
       this.options.routeHandler(fragment) :
       false;
   }
+
+  _createOrigin(protocol: string, hostname: string, port: string) {
```

Comment:
That's how I wrote it originally until I realized there wasn't a good way
to test with karma. I suppose the best thing to do in this case is to mock
Aurelia-PAL minimally with the location object (specifying specific values
for protocol, location, and hostname for multiple tests). This would allow
us to test the public method with multiple values.

However, I'm not sure if that actually solves the specific issue here, as
the expected value really depends on the browser and the current location.
If this is adequate, though, let's go for it.

2016-02-18 14:15 GMT-05:00 Bryan Smith <notifications@github.com>:

> In src/index.js
> <https://github.com/aurelia/history-browser/pull/22#discussion_r53366501>:
>
> > @@ -209,6 +216,10 @@ export class BrowserHistory extends History {
> >        this.options.routeHandler(fragment) :
> >        false;
> >    }
> > +
> > +  _createOrigin(protocol: string, hostname: string, port: string) {
>
> Yes, but I guess I don't see much value in testing the string
> concatenation. I'd be more concerned with testing the public method to
> verify that we're actually returning the expected value, given some
> combinations of location and history root.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aurelia/history-browser/pull/22/files#r53366501>.
>